src/core/qgspallabeling.cpp

```diff
@@ -3944,7 +3944,7 @@ QgsGeometry* QgsPalLabeling::prepareGeometry( const QgsGeometry* geometry, QgsRe
     clonedGeometry.reset( geom );
   }
 
-  if ( clipGeometry && !clipGeometry->contains( geom ) )
+  if ( clipGeometry && !clipGeometry->boundingBox().contains( geom->boundingBox() ) )
```

Comment:
I think this line should be left untouched. If a map is rotated then the extent geometry will be a rotated rectangle, so taking its bounding box will result in a larger extent then is actually visible. In this case we'd still need to clip. The geometryRequiresPreparation change is safe since it's already returned true if the map is rotated, but I think this second check should be left at the original version. What do you think?